addOns/client/src/main/java/org/zaproxy/addon/client/ClientIntegrationAPI.java

```diff
@@ -137,6 +142,15 @@ public ApiResponse handleApiAction(String name, JSONObject params) throws ApiExc
                         this.getParam(params, PARAM_STATEMENT_JSON, ""));
                 case ACTION_REPORT_ZEST_SCRIPT -> this.extension.addZestStatement(
                         this.getParam(params, PARAM_SCRIPT_JSON, ""));
+                case ACTION_CLIENT_MAP -> {
+                    String exportPath = this.getParam(params, PARAM_EXPORT_PATH, "");
+                    if (exportPath.isEmpty()) {
+                        throw new ApiException(
+                                ApiException.Type.MISSING_PARAMETER, PARAM_EXPORT_PATH);
+                    }
+                    this.extension.exportClientMap(exportPath);
+                    this.extension.addZestStatement(exportPath);
```

Comment:
This should be removed.

Reply:
`this.extension.exportClientMap(exportPath);` should be modified to return a boolean indicating success or not.
As @thc202 mentioned above it should also skip the display of the GUI dialog on failures when called from the API.